frontend/src/Onboarding/TransferCoursesScreen.tsx

```diff
@@ -35,253 +30,159 @@ const TitleText = styled.div`
   font-weight: 500;
 `;
 
-/**
- * Flattens the Requirement[] into only a list of Requirements/Requirement sets
- * This means that all inner lists will only contain one class or a list of the primary class and its labs/recitations
- * @param reqs
- */
-function flatten(reqs: Requirement[]): IRequiredCourse[][] {
-  return reqs.map(flattenOne).reduce((array, cur) => array.concat(cur), []);
-}
-
-function flattenOne(req: Requirement): IRequiredCourse[][] {
-  if (req.type === "COURSE") {
-    return [[req as IRequiredCourse]];
-  } else if (
-    req.type === "AND" &&
-    req.courses.filter(c => c.type === "COURSE").length
-  ) {
-    return [req.courses as IRequiredCourse[]];
-  } else if (req.type === "AND" || req.type === "OR") {
-    return flatten(req.courses);
-  } else {
-    return [];
-  }
-}
-
-interface TransferCoursesScreenProps {
+type TransferCoursesScreenProps = {
```

Comment:
Since I was refactoring anyway, I converted this to be a functional component as well, lmk thoughts on this

Reply:
awesome!!!